mtgrasp.smk

```diff
@@ -244,22 +244,25 @@ rule pre_polishing:
           bf = bf_sealer(params.r1, params.r2, wildcards.library, params.threads, params.sealer_fpr,params.k)
           count = sum(1 for line in open(input[0]))
           k = k_string_converter(params.k)
-          if check_gaps(input.target) != 0:
+          # check if input file is empty
+          if count == 0:
+                print(f"Input file {input.target} is empty, no mitochondrial sequence found.")
+                exit(1)
+          if check_gaps(input.target) != 0 and count == 2:
```

Comment:
Here, you could have 
```
num_gaps = check_gaps(input.target)
```
then, use that value for the subsequent checks. Just prevents having to keep reading the file unnecessarily.